### ISSUE TITLE
getComputedStyle's return value is not in sync after updating an existing <style> element

### ISSUE THREAD

**Issue by dirtyhenry** [jsdom/jsdom#2292]
### Basic info:

- **Node.js version: 8.11.1** 
- **jsdom version: 10.1.0** 

### Minimal reproduction case

```js
const { JSDOM } = require("jsdom");

const dom = new JSDOM(`<html>
<head>
  <style data-styled-components="">
    .myClass {
      font-size: 1.5em;
    }
  </style>
</head>
<body>
  <div>
    <h1 class="myClass">Hello World</h1>
  </div>
</body>

</html>
`);

const injectCss = document => {
  const el = document.createElement("style");
  el.setAttribute('id', 'myCSS');
  el.innerHTML = ".myClass { font-size: 2em; }";
  document.head.appendChild(el);
  return el;
};

const updateCss = (element) => {
  element.innerHTML = ".myClass { font-size: 3em; }";
};

const { window } = dom;
const { document } = window;
console.log(window.getComputedStyle(dom.window.document.querySelector("h1")).fontSize);
// 1.5em ✅
const myCSS = injectCss(document);
console.log(window.getComputedStyle(dom.window.document.querySelector("h1")).fontSize);
// 2em ✅
updateCss(myCSS);
console.log(window.getComputedStyle(dom.window.document.querySelector("h1")).fontSize);
// 2em 🛑 => it should be 3em
```

For search engine purposes and other users, I should mention this minimal code aims to reproduce what's going on using `react-testing-library` with `styled-components`. The CSS styles are built incrementally and the latest style of an element is not made accessible via `getComputedStyle`.

### How does similar code behave in browsers?

Cf. https://codepen.io/dirtyhenry/pen/mjJVNR

The updated style is taken into account.

**Comment by domenic** [comment]
Nice find, and great repro.

**Comment by artooras** [comment]
I just ran into this issue as well, so expressing my support for kicking up its priority. Since, if I understand correctly based on [documentation](https://github.com/gnapse/jest-dom#tohavestyle), `toHaveStyle` needs to match _all_ CSS for the test to pass, if I want to check for a single style attribute, I need to do it manually via `getComputedStyle`. So, this bug prevents me from being able to do that as well.

**Comment by privard** [comment]
Having the same problem here.

**Comment by domenic** [comment]
This code run on current master works, likely due to the fixes in 11.6.1 and 11.6.2. I will add a regression test to close the issue.
